src/api/modelTypes.js

```diff
@@ -655,63 +674,76 @@ export type Submessage = $ReadOnly<{|
  * `Message`: `PmMessage` and `StreamMessage`.
  */
 type MessageBase = $ReadOnly<{|
-  /** Our own flag; if true, really type `Outbox`. */
-  isOutbox: false,
-
-  /**
-   * These don't appear in `message` events, but they appear in a `/message`
-   * response when a search is involved.
-   */
-  match_content?: string,
-  match_subject?: string,
-
-  /**
-   * The `flags` story is a bit complicated:
-   *  * Absent in `event.message` for a `message` event... but instead
-   *    (confusingly) there is an `event.flags`.
-   *  * Present under `EVENT_NEW_MESSAGE`.
-   *  * Present under `MESSAGE_FETCH_COMPLETE`, and in the server `/message`
-   *    response that action is based on.
-   *  * Absent in the Redux `state.messages`; we move the information to a
-   *    separate subtree `state.flags`.
-   */
-  flags?: $ReadOnlyArray<string>,
-
-  /**
-   * Present under EVENT_NEW_MESSAGE and under MESSAGE_FETCH_COMPLETE,
-   * and in `state.messages`, all as an AvatarURL, because we
-   * translate into that form at the edge.
-   *
-   * For how it appears at the edge (and how we translate) see
-   * AvatarURL.fromUserOrBotData.
-   */
-  avatar_url: AvatarURL,
-
-  // The rest are believed to really appear in `message` events.
+  //
+  // Properties on `event.message` in `message` events. Order follows the
+  // doc. See comment on Message for how up-to-date these are.
+  //
 
   client: string,
   content: string,
   content_type: 'text/html',
   edit_history: $ReadOnlyArray<MessageEdit>,
-  gravatar_hash: string,
```

Comment:
modelTypes: Remove gravatar_hash from Message, deprecated in 2013 (really?)
    
    See zulip/zulip@9f7236eec, in 2017-05, which says this had been
    "deprecated and replaced" four years previously.

:smile: 

```
commit 9f7236eec1c16fe8333824612520b410d3a00046
Author: Tim Abbott <tabbott@zulip.com>
Date:   Tue May 9 22:20:47 2017 -0700

    message: Remove unused old gravatar_hash field from message dicts.
    
    This was deprecated and replaced some 4 years ago.
…
-            gravatar_hash     = gravatar_hash(sender_email), # Deprecated June 2013
             avatar_url        = avatar_url,
```

And following with `git log -G gravatar_hash` finds: zulip/zulip@d740d7c08274859b43a10899dd1f4cd45df76f31, which introduced `avatar_url`, with summary line "Show user-uploaded avatars on the website."

So yeah, apparently it was deprecated with the introduction of avatars that *weren't* from Gravatar and instead were avatars you uploaded to Zulip itself.

(Being originally gravatar-only sounds a bit less unreasonable when you realize that Gravatar isn't just about autogenerated avatars -- it's meant to be a site where you upload an avatar and then have it work across the Web.)